src/consumer/fetchManager.js

```diff
@@ -0,0 +1,107 @@
+const flatten = require('../utils/flatten')
+const flatMap = require('../utils/flatMap')
+const {
+  events: { FETCH_START, FETCH },
+} = require('./instrumentationEvents')
+
+const { values } = Object
+
+const fetchManager = ({ instrumentationEmitter, nodeIds, fetch, concurrency = 1 }) => {
+  const fetchers = {}
+  const inProgress = {}
+  const queues = Array(concurrency)
+    .fill()
+    .reduce((acc, _, runnerId) => ({ ...acc, [runnerId]: [] }), {})
+  let assignments = {}
+  let error
+
+  const fetchNode = async nodeId => {
+    if (fetchers[nodeId]) return fetchers[nodeId]
+
+    fetchers[nodeId] = (async () => {
+      const startFetch = Date.now()
+      instrumentationEmitter.emit(FETCH_START, { nodeId })
```

Comment:
If we're gonna include the `nodeId` in the FETCH_START and `FETCH` events, we gotta update the Typescript types and documentation to reflect that: 

https://github.com/tulios/kafkajs/blob/1876abcb5effd6183c1fbaa50dee9cef269b67a5/types/index.d.ts#L916-L917

https://github.com/tulios/kafkajs/blob/1876abcb5effd6183c1fbaa50dee9cef269b67a5/types/index.d.ts#L817-L820